File: top-interview-questions/two-sum.js
// https://leetcode.com/submissions/detail/530279400/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {

	let a = 0, b = 1;

	while (a < nums.length - 1) {
		b = a + 1;
		if (nums[a] + nums[b] === target && a != b) {
			return [a, b];
		};
		while (b < nums.length) {
			if (nums[a] + nums[b] === target && a != b) {
				return [a, b];
			};
			b++;
		};
		a++;
	};

	return [a, b];
};

File: problems/two-sum/two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] [Two Sum](https://leetcode.com/problems/two-sum/)
# https://neetcode.io/problems/two-integer-sum

# @lc code=start
from typing import List


class Solution:
    def properlyTypedTwoSum(self, nums: List[int], target: int) -> tuple[int, int]:
        for index_a, value_a in enumerate(nums):
            for index_b, value_b in enumerate(nums):
                if index_a == index_b:
                    continue
                if value_a + value_b == target:
                    return (index_a, index_b)

        raise ValueError("no sum equal to target in provided nums")

    def twoSum(self, nums: List[int], target: int) -> List[int]:
        return list(self.properlyTypedTwoSum(nums, target))


# @lc code=end
if __name__ == "__main__":
    solution = Solution()
    result = solution.twoSum([3, 2, 4], 6)
    print(f"{result = }")
